// @pulumi/linode/getLkeVersions.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getLkeVersionsOutput = exports.getLkeVersions = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Provides details about the Kubernetes versions available for deployment to a Kubernetes cluster.
 * For more information, see the [Linode APIv4 docs](https://techdocs.akamai.com/linode-api/reference/get-lke-versions).
 *
 * ## Example Usage
 *
 * The following example shows how one might use this data source to access information about a Linode LKE Version.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as linode from "@pulumi/linode";
 *
 * const example = linode.getLkeVersions({});
 * export const exampleOutput = example;
 * export const exampleOutputFirstVersion = example.then(example => example.versions?.[0]);
 * ```
 *
 * The following example shows how one might use this data source to access information about a Linode LKE Version
 * with additional information about the Linode LKE Version's tier (`enterprise` or `standard`).
 *
 * > **_NOTE:_**  This functionality may not be currently available to all users and can only be used with v4beta.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as linode from "@pulumi/linode";
 *
 * const exampleEnterprise = linode.getLkeVersions({
 *     tier: "enterprise",
 * });
 * export const exampleEnterpriseOutput = exampleEnterprise;
 * export const exampleEnterpriseOutputFirstVersion = exampleEnterprise.then(exampleEnterprise => exampleEnterprise.versions?.[0]);
 * ```
 */
function getLkeVersions(args, opts) {
    args = args || {};
    opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts || {});
    return pulumi.runtime.invoke("linode:index/getLkeVersions:getLkeVersions", {
        "tier": args.tier,
        "versions": args.versions,
    }, opts);
}
exports.getLkeVersions = getLkeVersions;
/**
 * Provides details about the Kubernetes versions available for deployment to a Kubernetes cluster.
 * For more information, see the [Linode APIv4 docs](https://techdocs.akamai.com/linode-api/reference/get-lke-versions).
 *
 * ## Example Usage
 *
 * The following example shows how one might use this data source to access information about a Linode LKE Version.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as linode from "@pulumi/linode";
 *
 * const example = linode.getLkeVersions({});
 * export const exampleOutput = example;
 * export const exampleOutputFirstVersion = example.then(example => example.versions?.[0]);
 * ```
 *
 * The following example shows how one might use this data source to access information about a Linode LKE Version
 * with additional information about the Linode LKE Version's tier (`enterprise` or `standard`).
 *
 * > **_NOTE:_**  This functionality may not be currently available to all users and can only be used with v4beta.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as linode from "@pulumi/linode";
 *
 * const exampleEnterprise = linode.getLkeVersions({
 *     tier: "enterprise",
 * });
 * export const exampleEnterpriseOutput = exampleEnterprise;
 * export const exampleEnterpriseOutputFirstVersion = exampleEnterprise.then(exampleEnterprise => exampleEnterprise.versions?.[0]);
 * ```
 */
function getLkeVersionsOutput(args, opts) {
    args = args || {};
    opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts || {});
    return pulumi.runtime.invokeOutput("linode:index/getLkeVersions:getLkeVersions", {
        "tier": args.tier,
        "versions": args.versions,
    }, opts);
}
exports.getLkeVersionsOutput = getLkeVersionsOutput;
