// @pulumi/linode/provider.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Provider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The provider type for the linode package. By default, resources use package-wide configuration
 * settings, however an explicit `Provider` instance may be created and passed during resource
 * construction to achieve fine-grained programmatic control over provider settings. See the
 * [documentation](https://www.pulumi.com/docs/reference/programming-model/#providers) for more information.
 */
class Provider extends pulumi.ProviderResource {
    /**
     * Returns true if the given object is an instance of Provider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === "pulumi:providers:" + Provider.__pulumiType;
    }
    /**
     * Create a Provider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        {
            resourceInputs["apiCaPath"] = args?.apiCaPath;
            resourceInputs["apiVersion"] = (args?.apiVersion) ?? utilities.getEnv("LINODE_API_VERSION");
            resourceInputs["configPath"] = args?.configPath;
            resourceInputs["configProfile"] = args?.configProfile;
            resourceInputs["disableInternalCache"] = pulumi.output(args?.disableInternalCache).apply(JSON.stringify);
            resourceInputs["eventPollMs"] = pulumi.output(args?.eventPollMs).apply(JSON.stringify);
            resourceInputs["lkeEventPollMs"] = pulumi.output(args?.lkeEventPollMs).apply(JSON.stringify);
            resourceInputs["lkeNodeReadyPollMs"] = pulumi.output(args?.lkeNodeReadyPollMs).apply(JSON.stringify);
            resourceInputs["maxRetryDelayMs"] = pulumi.output(args?.maxRetryDelayMs).apply(JSON.stringify);
            resourceInputs["minRetryDelayMs"] = pulumi.output(args?.minRetryDelayMs).apply(JSON.stringify);
            resourceInputs["objAccessKey"] = args?.objAccessKey;
            resourceInputs["objBucketForceDelete"] = pulumi.output(args?.objBucketForceDelete).apply(JSON.stringify);
            resourceInputs["objSecretKey"] = args?.objSecretKey ? pulumi.secret(args.objSecretKey) : undefined;
            resourceInputs["objUseTempKeys"] = pulumi.output(args?.objUseTempKeys).apply(JSON.stringify);
            resourceInputs["skipImplicitReboots"] = pulumi.output(args?.skipImplicitReboots).apply(JSON.stringify);
            resourceInputs["skipInstanceDeletePoll"] = pulumi.output(args?.skipInstanceDeletePoll).apply(JSON.stringify);
            resourceInputs["skipInstanceReadyPoll"] = pulumi.output(args?.skipInstanceReadyPoll).apply(JSON.stringify);
            resourceInputs["token"] = args?.token;
            resourceInputs["uaPrefix"] = (args?.uaPrefix) ?? utilities.getEnv("LINODE_UA_PREFIX");
            resourceInputs["url"] = (args?.url) ?? utilities.getEnv("LINODE_URL");
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["objSecretKey"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Provider.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * This function returns a Terraform config object with terraform-namecased keys,to be used with the Terraform Module Provider.
     */
    terraformConfig() {
        return pulumi.runtime.call("pulumi:providers:linode/terraformConfig", {
            "__self__": this,
        }, this);
    }
}
exports.Provider = Provider;
/** @internal */
Provider.__pulumiType = 'linode';
