// @pulumi/linode/vpcSubnet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcSubnet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Manages a Linode VPC subnet.
 * For more information, see the [Linode APIv4 docs](https://techdocs.akamai.com/linode-api/reference/post-vpc-subnet).
 *
 * ## Example Usage
 *
 * Create a VPC subnet:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as linode from "@pulumi/linode";
 *
 * const test = new linode.VpcSubnet("test", {
 *     vpcId: 123,
 *     label: "test-subnet",
 *     ipv4: "10.0.0.0/24",
 * });
 * ```
 *
 * Create a VPC subnet with an implicitly determined IPv6 range:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as linode from "@pulumi/linode";
 *
 * const testVpc = new linode.Vpc("test", {
 *     label: "test-vpc",
 *     region: "us-mia",
 *     ipv6s: [{
 *         range: "/52",
 *     }],
 * });
 * // NOTE: IPv6 VPCs may not currently be available to all users.
 * const test = new linode.VpcSubnet("test", {
 *     vpcId: testVpc.id,
 *     label: "test-subnet",
 *     ipv4: "10.0.0.0/24",
 *     ipv6s: [{
 *         range: "auto",
 *     }],
 * });
 * ```
 *
 * ## IPv6
 *
 * > **Limited Availability** IPv6 VPCs may not currently be available to all users.
 *
 * The following arguments can be configured for each entry under the `ipv6` field:
 *
 * * `range` - (Optional) An existing IPv6 prefix owned by the current account or a forward slash (/) followed by a valid prefix length. If `auto`, a range with the default prefix will be allocated for this VPC.
 *
 * * `allocatedRange` - (Read-Only) The value of range computed by the API. This is necessary when needing to access the range for an implicit allocation.
 *
 * ## Import
 *
 * Linode Virtual Private Cloud (VPC) Subnet can be imported using the `vpc_id` followed by the subnet `id` separated by a comma, e.g.
 *
 * ```sh
 * $ pulumi import linode:index/vpcSubnet:VpcSubnet my_subnet_duplicated 1234567,7654321
 * ```
 */
class VpcSubnet extends pulumi.CustomResource {
    /**
     * Get an existing VpcSubnet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VpcSubnet(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of VpcSubnet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpcSubnet.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["created"] = state?.created;
            resourceInputs["ipv4"] = state?.ipv4;
            resourceInputs["ipv6s"] = state?.ipv6s;
            resourceInputs["label"] = state?.label;
            resourceInputs["linodes"] = state?.linodes;
            resourceInputs["updated"] = state?.updated;
            resourceInputs["vpcId"] = state?.vpcId;
        }
        else {
            const args = argsOrState;
            if (args?.label === undefined && !opts.urn) {
                throw new Error("Missing required property 'label'");
            }
            if (args?.vpcId === undefined && !opts.urn) {
                throw new Error("Missing required property 'vpcId'");
            }
            resourceInputs["ipv4"] = args?.ipv4;
            resourceInputs["ipv6s"] = args?.ipv6s;
            resourceInputs["label"] = args?.label;
            resourceInputs["vpcId"] = args?.vpcId;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["linodes"] = undefined /*out*/;
            resourceInputs["updated"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VpcSubnet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VpcSubnet = VpcSubnet;
/** @internal */
VpcSubnet.__pulumiType = 'linode:index/vpcSubnet:VpcSubnet';
