// @pulumi/linode/config/vars.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
const __config = new pulumi.Config("linode");
Object.defineProperty(exports, "apiCaPath", {
    get() {
        return __config.get("apiCaPath");
    },
    enumerable: true,
});
Object.defineProperty(exports, "apiVersion", {
    get() {
        return __config.get("apiVersion") ?? utilities.getEnv("LINODE_API_VERSION");
    },
    enumerable: true,
});
Object.defineProperty(exports, "configPath", {
    get() {
        return __config.get("configPath");
    },
    enumerable: true,
});
Object.defineProperty(exports, "configProfile", {
    get() {
        return __config.get("configProfile");
    },
    enumerable: true,
});
Object.defineProperty(exports, "disableInternalCache", {
    get() {
        return __config.getObject("disableInternalCache");
    },
    enumerable: true,
});
Object.defineProperty(exports, "eventPollMs", {
    get() {
        return __config.getObject("eventPollMs");
    },
    enumerable: true,
});
Object.defineProperty(exports, "lkeEventPollMs", {
    get() {
        return __config.getObject("lkeEventPollMs");
    },
    enumerable: true,
});
Object.defineProperty(exports, "lkeNodeReadyPollMs", {
    get() {
        return __config.getObject("lkeNodeReadyPollMs");
    },
    enumerable: true,
});
Object.defineProperty(exports, "maxRetryDelayMs", {
    get() {
        return __config.getObject("maxRetryDelayMs");
    },
    enumerable: true,
});
Object.defineProperty(exports, "minRetryDelayMs", {
    get() {
        return __config.getObject("minRetryDelayMs");
    },
    enumerable: true,
});
Object.defineProperty(exports, "objAccessKey", {
    get() {
        return __config.get("objAccessKey");
    },
    enumerable: true,
});
Object.defineProperty(exports, "objBucketForceDelete", {
    get() {
        return __config.getObject("objBucketForceDelete");
    },
    enumerable: true,
});
Object.defineProperty(exports, "objSecretKey", {
    get() {
        return __config.get("objSecretKey");
    },
    enumerable: true,
});
Object.defineProperty(exports, "objUseTempKeys", {
    get() {
        return __config.getObject("objUseTempKeys");
    },
    enumerable: true,
});
Object.defineProperty(exports, "skipImplicitReboots", {
    get() {
        return __config.getObject("skipImplicitReboots");
    },
    enumerable: true,
});
Object.defineProperty(exports, "skipInstanceDeletePoll", {
    get() {
        return __config.getObject("skipInstanceDeletePoll");
    },
    enumerable: true,
});
Object.defineProperty(exports, "skipInstanceReadyPoll", {
    get() {
        return __config.getObject("skipInstanceReadyPoll");
    },
    enumerable: true,
});
Object.defineProperty(exports, "token", {
    get() {
        return __config.get("token");
    },
    enumerable: true,
});
Object.defineProperty(exports, "uaPrefix", {
    get() {
        return __config.get("uaPrefix") ?? utilities.getEnv("LINODE_UA_PREFIX");
    },
    enumerable: true,
});
Object.defineProperty(exports, "url", {
    get() {
        return __config.get("url") ?? utilities.getEnv("LINODE_URL");
    },
    enumerable: true,
});
